src/bin/outpack_query.rs

```diff
@@ -1,39 +1,76 @@
 extern crate core;
 
 use getopts::Options;
-use std::{env, process};
+use std::{env, process::ExitCode};
+
+use outpack::query::QueryError;
 
 fn print_usage(program: &str, opts: Options) {
     let brief = format!("Usage: {} [options]", program);
     print!("{}", opts.usage(&brief));
 }
 
-fn parse_args(args: &[String]) -> (String, String) {
+enum Args {
+    Parse { query: String },
+    Eval { query: String, root: String },
+}
+
+fn parse_args(args: &[String]) -> Args {
```

Comment:
We've been chatting about using [clap](https://docs.rs/clap/latest/clap/) for our CLI interface, ticket here https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-4553/Use-clap-for-the-outpack-rust-CLI . Something might be worth having a look at some point down the line if it looks interesting?

Reply:
Yes I saw that ticket, getopts is pretty limited so I think that would be a great idea. I kept the change to this file as low-effort as possible knowing it would probably get re-written soon.